### PR TITLE
continuoustest: Query with strong read consistency enabled

### PR DESCRIPTION
**Pull request by narqo**
#### What this PR does

Related to experimental ingest storage, this PR updates mimir-continuous-test to query with read-after-write (strong read consistency) enabled.

#### Which issue(s) this PR fixes or relates to

n/a

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: pkg/continuoustest/client.go [hunk 1]
```diff
@@ -153,6 +154,8 @@ func (c *Client) QueryRange(ctx context.Context, query string, start, end time.T
 	ctx, cancel := context.WithTimeout(ctx, c.cfg.ReadTimeout)
 	defer cancel()
 
+	ctx = querierapi.ContextWithReadConsistency(ctx, querierapi.ReadConsistencyStrong)
```
Comment (pracucci) [inline pkg/continuoustest/client.go:157]: The ctx is passed to the read client, which is just an HTTP client. The context will not be propagated, so this won't work. We should inject the HTTP Header instead (`X-Read-Consistency: strong`). I think we can easily do it configuring a Roundtripper on the HTTP client passed to the Prometheus client in `NewClient()`.

**Review by pracucci — COMMENTED** [review]
Thanks for working on this! I've left a comment.

File: pkg/continuoustest/client.go (hunk 1 above)
Comment (narqo) [reply pkg/continuoustest/client.go:157]: I believe we handle that (extracting the consistency level into the header) in the `clientRoundTripper` below, don't we?

Comment (pstibrany) [reply pkg/continuoustest/client.go:157]: You're right @narqo and updated unit tests also show that.

**Review by pstibrany — APPROVED** [review]

File: pkg/continuoustest/client.go (hunk 1 above)
Comment (pracucci) [reply pkg/continuoustest/client.go:157]: > I believe we handle that (extracting the consistency level from context into the header) in the clientRoundTripper chunk below, don't we?

Sorry. I totally missed it.

**Pull request merged**